protocol/messenger_communities_import_discord.go

```diff
@@ -412,6 +414,7 @@ func (m *Messenger) startPublishImportProgressInterval(c chan *discord.ImportPro
 	var currentProgress *discord.ImportProgress
 
 	go func() {
+		defer gocommon.LogOnPanicAndRethrow()
 		ticker := time.NewTicker(2 * time.Second)
 		defer ticker.Stop()
```

Comment:
Go uses a Last In First Out stack for multiple `defers`, so if you want the panic logger handler to be executed first it will need to be called last.

Which does mean the regex will need another iteration.

Reply:
Here's a nice example of the `defer` stack in practice https://golangbyexample.com/multiple-defer-functions-golang/ .

What do you think? Should `LogOnPanicAndRethrow()` be called first in the defer order?

Reply:
But I think we want the panic recover to be called the **last**, no 🤔 
Like if another panic happens during `ticker.Stop()`, we still want to catch it after.

Reply:
@igor-sirotin I've just checked how panics effect `defer` and all defers will be triggered before a panic breaks a function. I thought that a second panic within a defer would break that defer and then the function. But that isn't how it works. All the panics from the same function are queued for recovery, all defers will be called regardless of how many defers panic.

So my presumption was wrong.

Reply:
@qfrank and @igor-sirotin 

I may be have only been half wrong 😆 check this code: https://go.dev/play/p/A2ik-YSGsWW

```go
package main

import (
	"fmt"
	"runtime/debug"
)

func logPanic() {
	if err := recover(); err != nil {
		fmt.Println("panic in goroutine", "error", err, "stacktrace", string(debug.Stack()))
		panic(err)
	}
}

func main() {
	defer logPanic()
	defer func() {
		fmt.Println("defer 1")
		panic(" defer 1")
	}()
	defer func() {
		fmt.Println("defer 2")
		panic("defer 2")
	}()
	fmt.Println("main start")
	panic("main")
	fmt.Println("main end")
}
```

The output is:

```
panic: main
	panic: defer 2
	panic:  defer 1 [recovered]
	panic:  defer 1

goroutine 1 [running]:
main.logPanic()
	/tmp/sandbox2687859731/prog.go:11 +0x11c
panic({0x498700?, 0x4d3e38?})
	/usr/local/go-faketime/src/runtime/panic.go:785 +0x132
main.main.func1()
	/tmp/sandbox2687859731/prog.go:19 +0x59
panic({0x498700?, 0x4d3e48?})
	/usr/local/go-faketime/src/runtime/panic.go:785 +0x132
main.main.func2()
	/tmp/sandbox2687859731/prog.go:23 +0x59
panic({0x498700?, 0x4d3e18?})
	/usr/local/go-faketime/src/runtime/panic.go:785 +0x132
main.main()
	/tmp/sandbox2687859731/prog.go:26 +0xa5

Program exited.
```

Only the last panic is recovered and logged

Reply:
TIL: When a deferred function itself calls panic, it effectively overwrites any existing panic that was already in progress. This is because, at any given moment, Go can only handle one active panic state per goroutine.